auth: fix panic using WithRoot and improve JWT coverage

hexfusion opened the pull request:
ref: https://github.com/coreos/etcd/issues/9695

hexfusion commented:
@gyuho FYI

https://jenkins-etcd-public.prod.coreos.systems/job/etcd-ci-ppc64/4075/console
`fatal: write error: No space left on device`

gyuho commented:
@hexfusion 

@mkumatag Can you clean up the disk :)

gyuho commented:
@hexfusion Can we add a simple unit test? Thanks!

hexfusion commented:
> @hexfusion Can we add a simple unit test? Thanks!

@gyuho yes sir will address

hexfusion commented:
@gyuho test added, thanks.

xiang90 commented on auth/store.go:
```diff
@@ -1284,7 +1284,7 @@ func NewTokenProvider(
 
 func (as *authStore) WithRoot(ctx context.Context) context.Context {
-	if !as.IsAuthEnabled() {
+	if tt := tokenProviderType(as); tt != "simple" || !as.IsAuthEnabled() {
```
we probably should define "simple" as a const.

hexfusion replied:
@xiang90 good idea, done.

hexfusion commented:
Updated the check and added some testing. In general, it is very awkward how NewTokenProvider accepts opts as a string, that is fine for simple but not for jwt. I didn't address this here but this should be looked at further. I also removed the util function and tests as they were not necessary.

/cc @gyuho @joelegasse

hexfusion commented:
@gyuho @joelegasse @xiang90 thank you!
